src/snapshot_manager.rs

```diff
@@ -0,0 +1,76 @@
+use parking_lot::RwLock;
+use std::{
+    collections::HashMap,
+    sync::{atomic::AtomicU64, Arc},
+};
+
+use crate::{
+    db_state::CoreDbState, error::SlateDBError, manifest::Manifest, manifest_store::ManifestStore,
+};
+
+pub struct SnapshotManager {
+    snapshots: RwLock<HashMap<u64, Snapshot>>,
+    next_snapshot_id: AtomicU64,
+    manifest_store: Arc<ManifestStore>,
+}
+
+#[derive(Copy, Clone)]
+pub struct Snapshot {
+    id: u64,
+    pub(crate) manifest_id: u64,
+}
+
+impl SnapshotManager {
+    pub fn new(manifest_store: Arc<ManifestStore>) -> Self {
+        Self {
+            snapshots: RwLock::new(HashMap::new()),
+            next_snapshot_id: AtomicU64::new(0),
+            manifest_store,
+        }
+    }
+
+    pub async fn create_snapshot(&self, core: CoreDbState) -> Result<Snapshot, SlateDBError> {
+        let next_id = self
```

Comment:
I was thinking we would use a random u64 and verify that it's not already in the snapshot. Multiple clients can all create read snapshots to a single DB manifest, so incremental IDs won't work so well.